pkg/catalog/outbound_traffic_policies.go

```diff
@@ -292,60 +193,27 @@ func (mc *MeshCatalog) getDestinationServicesFromTrafficTarget(t *access.Traffic
 	return destServices, nil
 }
 
-// GetWeightedClustersForUpstream returns Envoy cluster weights for the given
-// upstream service, the apex service of a TrafficSplit.
-func (mc *MeshCatalog) GetWeightedClustersForUpstream(upstream service.MeshService) []service.WeightedCluster {
-	var weightedClusters []service.WeightedCluster
-	apexServices := mapset.NewSet()
-
-	for _, split := range mc.meshSpec.ListTrafficSplits() {
-		// Split policy must be in the same namespace as the upstream service
-		if split.Namespace != upstream.Namespace {
-			continue
-		}
-		rootServiceName := k8s.GetServiceFromHostname(split.Spec.Service)
-		if rootServiceName != upstream.Name {
-			// This split policy does not correspond to the upstream service
-			continue
-		}
-
-		if apexServices.Contains(split.Spec.Service) {
-			log.Error().Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrMultipleSMISplitPerServiceUnsupported)).
-				Msgf("Skipping traffic split policy %s/%s as there is already a corresponding policy for apex service %s", split.Namespace, split.Name, split.Spec.Service)
-			continue
-		}
-
-		for _, backend := range split.Spec.Backends {
-			if backend.Weight == 0 {
-				// Skip backends with a weight of 0
-				log.Warn().Msgf("Skipping backend %s that has a weight of 0 in traffic split policy %s/%s", backend.Service, split.Namespace, split.Name)
-				continue
-			}
-			backendCluster := service.WeightedCluster{
-				ClusterName: service.ClusterName(split.Namespace + "/" + backend.Service),
-				Weight:      backend.Weight,
-			}
-			weightedClusters = append(weightedClusters, backendCluster)
-		}
-		apexServices.Add(split.Spec.Service)
-	}
-
-	return weightedClusters
-}
-
-// ListMeshServicesForIdentity returns a list of services the service with the
-// given identity can communicate with, including apex TrafficSplit services.
-func (mc *MeshCatalog) ListMeshServicesForIdentity(identity identity.ServiceIdentity) []service.MeshService {
-	upstreamServices := mc.ListOutboundServicesForIdentity(identity)
+// listAllowedUpstreamServicesIncludeVirtual returns a list of services the given downstream service identity
+// is authorized to communicate with, including virtual services (traffic split apex) that are not backed by
+// pods.
+func (mc *MeshCatalog) listAllowedUpstreamServicesIncludeVirtual(downstreamIdentity identity.ServiceIdentity) []service.MeshService {
```

Comment:
Could we come up with another name instead of Virtual ? I feel like we are introducing another term that you need to be aware of and its not intuitive,  maybe : 
```suggestion
func (mc *MeshCatalog) listAllowedUpstreamServicesIncludeApex(downstreamIdentity identity.ServiceIdentity) []service.MeshService {
```